enterprise-suite/scripts/lbc.py

```diff
@@ -284,13 +293,91 @@ def check_resource_list(cmd, expected, fail_msg):
         return all_found
     return False
 
-# Checks for console PVCs
-def are_pvcs_created(namespace):
-    return check_resource_list(
-        cmd='kubectl get pvc --namespace={} --no-headers'.format(namespace),
-        expected=CONSOLE_PVCS,
-        fail_msg='Found some PVCs ({}) from previous console install, but not all expected: {}.\nTo avoid data loss, please remove them manually'
-    )
+# Returns two lists.  For each of the PVC types we care about, the associated PV
+# is in the first or second list if the reclaim policy is RETAIN or not respectively.
+# Each list element is a tuple of PV name and claim.
+def pvsRetainedOrNot(namespace):
+    returncode, stdout, _ = run('kubectl --namespace {} get pv --no-headers'
+                             .format(namespace))
+    retained = []
+    notRetained = []
+
+    if returncode != 0:
+        printerr("Unable to retrieve PV info.  Proceed with caution.")
+    else:
+        for pvc in CONSOLE_PVCS:
+            # We might want to do this by parsing yaml instead...
+            # pvc-accccc27-2ef8-11e9-a408-080027c68b7b   32Gi       RWO            Delete           Bound    lightbend/alertmanager-storage   standard                24h
+            claim = namespace + '/' + pvc
+
+            match = re.search(r'^([\w-]+)\s+\w+\s+\w+\s+(\w+)\s+Bound\s+('+claim+r')\s', stdout, re.MULTILINE)
```

Comment:
Instead of parsing the human friendly output with regex, I think it would be much better to use kubectl's templating system. E.g. you can do something like

```
kubectl get pv -o go-template='{{ range .items }}{{ printf "%s\n" .metadata.name }}{{ end }}'
```

Reply:
Yeah.  Figured there was a better way but my Go template foo is yellow-belt at best.  Changed.